tools/testing/target_determination/heuristics/interface.py

```diff
@@ -41,24 +56,84 @@ def integrate_priorities(self, other: "TestPrioritizations") -> None:
+            self.highly_relevant,
+            other.highly_relevant,
+            exclude_tests=[],
+            orig_tests=orig_tests,
         )
 
         self.probably_relevant = TestPrioritizations._merge_tests(
             self.probably_relevant,
             other.probably_relevant,
             exclude_tests=self.highly_relevant,
+            orig_tests=orig_tests,
+        )
+
+        # Remove any tests that are now in a higher priority list
+        self.unranked_relevance = TestPrioritizations._merge_tests(
+            self.unranked_relevance,
+            new_tests=[],
+            exclude_tests=(self.highly_relevant + self.probably_relevant),
+            orig_tests=orig_tests,
         )
 
-        # We don't expect every heuristics to list all unranked tests. Easier to compute the list from scratch
-        ranked_tests = self.highly_relevant + self.probably_relevant
-        self.unranked_relevance = [
-            t for t in self.unranked_relevance if t not in ranked_tests
-        ]
+    def find_conflicts(self) -> List[str]:
+        """
+        Returns a list of tests which are in multiple lists.
+        """
+        tests = self.highly_relevant + self.probably_relevant + self.unranked_relevance
+        return [test for test in tests if tests.count(test) > 1]
+
+    def get_unexpected_tests(self, expected: List[str]) -> List[str]:
+        """
+        Returns a list of tests which are not in the expected list.
+        """
+        tests = self.highly_relevant + self.probably_relevant + self.unranked_relevance
+        return [test for test in tests if test not in expected]
+
+    def print_info(self, expected: List[str]) -> None:
+        def _print_tests(label: str, tests: List[str]) -> None:
+            if not tests:
+                return
+
+            print(f"{label} tests ({len(tests)}):")
+            for test in tests:
+                if test in tests:
+                    print(f"  {test}")
+
+        _print_tests("Highly relevant", self.highly_relevant)
+        _print_tests("Probably relevant", self.probably_relevant)
+        _print_tests("Unranked relevance", self.unranked_relevance)
+
+        conflicts = self.find_conflicts()
+        if conflicts:
+            print(f"WARNING: {len(conflicts)} tests are in multiple lists:")
+            for test in conflicts:
+                lists = []
+                for list_name in ["highly_relevant", "probably_relevant", "unranked"]:
+                    if test in getattr(self, list_name):
+                        lists.append(list_name)
+                print(f"  {test} is in {' ,'.join(lists)}")
+        else:
+            print("No conflicts found")
+
+        unexpected = self.get_unexpected_tests(expected)
+        if unexpected:
+            print(f"WARNING: {len(unexpected)} tests are not in the expected list:")
+            for test in unexpected:
+                print(f"  {test}")
+        else:
+            print("No unexpected tests found")
```

Comment:
In what cases would you expect conflicts or unexpected tests?

Reply:
my apologies, I thought I'd deleted that code (it was for debugging). Turns out the files were sitting uncommitted on my box